AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
+		debug_printInfo("  MAIN: Property request, request_id:%s, status: %d", request_id_buf, property_response.status);
+	}
+	if (payload == NULL)
+	{
+		debug_printWarn("  MAIN: NULL payload");
+	}
+	else
+	{
+		handle_property_message(az_span_create_from_str((char*)payload), &property_response);
+	}
+}
 
+/**********************************************
+* Callback for device twin patch
+**********************************************/
+void receivedFromCloud_patch(uint8_t* topic, uint8_t* payload)
+{
+	debug_printInfo("  MAIN: Device Property Patch");
+
+	if (topic == NULL)
+	{
+		debug_printWarn("  MAIN: NULL topic");
+		return;
+	}
+
+	az_iot_pnp_client_property_response property_response;
+	az_span property_topic = az_span_create_from_str((char *)topic);
+	az_result rc = az_iot_pnp_client_property_parse_received_topic(&pnp_client, property_topic, &property_response);
+	RETURN_WITH_MESSAGE_IF_FAILED(rc, "az_iot_pnp_client_property_parse_received_topic failed");
+
+	if (payload == NULL)
+	{
+		debug_printWarn("  MAIN: NULL payload");
+	}
+	else
+	{
+		handle_property_message(az_span_create_from_str((char*)payload), &property_response);
+	}
+}
+
+/**********************************************
+* Check if button(s) was pressed.
+**********************************************/
+void check_button_status(void)
+{
+	az_span button_event_span = AZ_SPAN_FROM_BUFFER(button_event_payload);
+	az_json_writer jw;
+
+	// save flags in case user pressed buttons very fast, and for error case.
+	bool sw0 = button_press_data.flag.sw0 == 1 ? true : false;
+	bool sw1 = button_press_data.flag.sw1 == 1 ? true : false;
+
+	// clear the flags
+	button_press_data.flag.AsUSHORT = 0;
+
+	if (sw0 != true && sw1 != true)
+	{
+		return;
+	}
+
+	RETURN_IF_FAILED(start_json_object(&jw, button_event_span));
+
+	if (sw0 == true)
+	{
+		debug_printGood("  MAIN: Button SW0 Press Count %lu", button_press_data.sw0_press_count);
+		RETURN_IF_FAILED(append_button_press_telemetry(&jw, span_event_name_button_sw0, button_press_data.sw0_press_count));
+	}
+
+	if (sw1 == true)
+	{
+		debug_printGood("  MAIN: Button SW1 Press Count %lu", button_press_data.sw1_press_count);
+		RETURN_IF_FAILED(append_button_press_telemetry(&jw, span_event_name_button_sw1, button_press_data.sw1_press_count));
+	}
+
+	RETURN_IF_FAILED(end_json_object(&jw));
+
+	button_event_span = az_json_writer_get_bytes_used_in_destination(&jw);
+	if (mqtt_publish_message(telemetry_topic, button_event_span, 0) != 0)
+	{
+		debug_printError("  MAIN: Failed to publish Button Press event telemetry");
+	}
+}
+
+/**********************************************
+* Check if LED status has changed or not.
+**********************************************/
+void check_led_status(twin_properties_t* twin_properties)
+{
+	twin_properties_t* twin_properties_ptr;
+	twin_properties_t twin_properties_local;
+
+	bool bForceSync = false; // force LED status synchronization if this request is coming from Twin Get
```

Comment:
nit: consistent code style
```suggestion
	bool b_force_sync = false; // force LED status synchronization if this request is coming from Twin Get
```

Reply:
done

[commit](https://github.com/daisukeiot/Microchip-PIC-IoT-Wx-1/commit/4150d1aef0b0106fd1b4799cdb5b2d4520083ebf)